internal/providers/terraform/azure/application_insights_web_t.go

```diff
@@ -1,51 +1,27 @@
 package azure
 
 import (
-	"strings"
-
-	"github.com/shopspring/decimal"
-	"github.com/tidwall/gjson"
-
+	"github.com/infracost/infracost/internal/resources/azure"
 	"github.com/infracost/infracost/internal/schema"
 )
 
-func GetAzureRMApplicationInsightsWebRegistryItem() *schema.RegistryItem {
+func getAzureRMApplicationInsightsWebRegistryItem() *schema.RegistryItem {
 	return &schema.RegistryItem{
 		Name:  "azurerm_application_insights_web_test",
-		RFunc: NewAzureRMApplicationInsightsWeb,
+		RFunc: NewApplicationInsightsWebTest,
 		ReferenceAttributes: []string{
 			"resource_group_name",
 		},
 	}
 }
-
-func NewAzureRMApplicationInsightsWeb(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
-	region := lookupRegion(d, []string{"resource_group_name"})
-	costComponents := []*schema.CostComponent{}
-
-	if d.Get("kind").Type != gjson.Null {
-		if strings.ToLower(d.Get("kind").String()) == "multistep" && d.Get("enabled").Type == gjson.True {
-			costComponents = append(costComponents, appInsightCostComponents(
-				region,
-				"Multi-step web test",
-				"test",
-				"Multi-step Web Test",
-				"Enterprise",
-				decimalPtr(decimal.NewFromInt(1))))
-		}
+func NewApplicationInsightsWebTest(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	r := &azure.ApplicationInsightsWebTest{Address: d.Address, Region: lookupRegion(d, []string{"resource_group_name"})}
+	if !d.IsEmpty("enabled") {
+		r.Enabled = d.Get("enabled").Bool()
 	}
-
-	if len(costComponents) == 0 {
-		return &schema.Resource{
-			Name:      d.Address,
-			IsSkipped: true,
-			NoPrice:   true,
-		}
-	}
-
-	return &schema.Resource{
-		Name:           d.Address,
-		CostComponents: costComponents,
+	if !d.IsEmpty("kind") {
```

Comment:
I don't think we need this `IsEmpty` check.

Reply:
Applied